From ad4e6764d7b7ae6ac40802ee2f7507a5e91de9ac Mon Sep 17 00:00:00 2001
From: Meghana G <meghanag@ami.com>
Date: Thu, 11 Jun 2026 17:29:38 +0000
Subject: [PATCH] Fix stdexec inline_scheduler namespace for newer exec headers

Newer stdexec/exec headers expose inline_scheduler under stdexec, not exec.
Update pldm platform-mc call sites accordingly.

Upstream-Status: Backport

Signed-off-by: Meghana G <meghanag@ami.com>
Signed-off-by: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
---
 platform-mc/sensor_manager.cpp   | 2 +-
 platform-mc/terminus_manager.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/platform-mc/sensor_manager.cpp b/platform-mc/sensor_manager.cpp
index 2ffc679dd..c828e9947 100644
--- a/platform-mc/sensor_manager.cpp
+++ b/platform-mc/sensor_manager.cpp
@@ -140,7 +140,7 @@ void SensorManager::doSensorPolling(pldm_tid_t tid)
                 rcOpt = PLDM_SUCCESS;
             }
         }),
-        exec::default_task_context<void>(exec::inline_scheduler{}));
+        exec::default_task_context<void>(stdexec::inline_scheduler{}));
 }
 
 exec::task<int> SensorManager::doSensorPollingTask(pldm_tid_t tid)
diff --git a/platform-mc/terminus_manager.cpp b/platform-mc/terminus_manager.cpp
index 56902adda..5fa8bf088 100644
--- a/platform-mc/terminus_manager.cpp
+++ b/platform-mc/terminus_manager.cpp
@@ -141,7 +141,7 @@ void TerminusManager::discoverMctpTerminus(const MctpInfos& mctpInfos)
     auto& [scope, rcOpt] = discoverMctpTerminusTaskHandle.emplace();
     scope.spawn(discoverMctpTerminusTask() |
                     stdexec::then([&](int rc) { rcOpt.emplace(rc); }),
-                exec::default_task_context<void>(exec::inline_scheduler{}));
+                exec::default_task_context<void>(stdexec::inline_scheduler{}));
 }
 
 TerminiMapper::iterator
-- 
2.43.0

